auth/default_authorizer.cc

```diff
@@ -303,6 +311,54 @@ future<> default_authorizer::revoke_all(const resource& resource) {
     });
 }
 
+future<> default_authorizer::revoke_all(const resource& resource) {
+    if (legacy_mode(_qp)) {
+        co_return co_await revoke_all_legacy(resource);
+    }
+    auto name = resource.name();
+    try {
+        // guard has to be taken before we execute select so that
+        // operation is linearizable with other auth statements
+        // in other words delete and select below operate on the same data
+        auto group0_guard = co_await _group0_client.start_operation(&_as);
+        auto timestamp = group0_guard.write_timestamp();
```

Comment:
We discussed this on a call, to make the code more consistent with `announce_mutations`, and to keep the logic of taking guards in a single place, the first guard should be taken inside `announce_mutations_with_batching` and passed to `mutations_generator` as a parameter.

Then `co_yield` should provide the latest guard every time, if `announce_mutations_with_batching` needs to drop and take another guard.

Also the loop over mutations after `get_mutations_internal` can be turned into an assertion that we have only one mutation because this is a single row delete. This way we guarantee that every mutation uses the correct timestamp from the correct guard provided to generate this mutation.

Reply:
done